File: examples/python/src/02_search.py
import json
from client import McpSseClient, env

def main():
    search_phrase = env("DEMO_SEARCH_PHRASE", "Ravensburger")
    c = McpSseClient(env("MCP_BASE_URL"))
    res = c.call_tool("searchProduct", {"searchPhrase": search_phrase, "limit": 20, "page": 1})
    print(json.dumps(res, indent=2))

if __name__ == "__main__":
    main()

File: examples/node/src/02_search.ts
import { createClient, getEnv } from "./_mcp_sse_client.js";

async function main() {
  const client = createClient();
  const searchPhrase = getEnv("DEMO_SEARCH_PHRASE", "Ravensburger");

  const res = await client.callTool("searchProduct", { searchPhrase, limit: 20, page: 1 });
  console.log(JSON.stringify(res, null, 2));

  client.close();
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
